annotto-front/src/modules/project/components/common/image/ZoomMark.js

```diff
@@ -0,0 +1,210 @@
+
+const ZoomMark = ({
+  index,
+  tasks,
+  annotationIndex,
+  predictionIndex,
+  zone,
+  value,
+  imageWidth,
+  imageHeight,
+  isSelected,
+  onValidateClick,
+  onDeleteClick,
+  onSelectClick,
+  onTransformEnd,
+  ...props
+}) => {
+  const lineRef = useRef()
+  const trRef = useRef()
+
+  const [isHovered, setIsHovered] = useState(false)
+
+  const [tagIcon] = useImage(`${process.env.PUBLIC_URL}/static/images/tag.svg`)
+  const [patternIcon] = useImage(`${process.env.PUBLIC_URL}/static/images/pattern.svg`)
+  const [closeIcon] = useImage(`${process.env.PUBLIC_URL}/static/images/close.svg`)
+  const [validateIcon] = useImage(`${process.env.PUBLIC_URL}/static/images/validate.svg`)
+
+  const isPrefill = isNumber(annotationIndex) && isNumber(predictionIndex)
+  const isPrediction = isNumber(predictionIndex) && !isNumber(annotationIndex)
+  const isEditable = !isPrediction && !isPrefill
+
+  const points = useMemo(
+    () => flatten(zone.map(({ x, y }) => [x * imageWidth, y * imageHeight])),
+    [zone, imageHeight, imageWidth]
+  )
+
+  const task = useMemo(() => {
+    const currentTask = tasks.find((t) => t.value === value)
+    if (currentTask && !currentTask?.color) {
+      currentTask.color = theme.colors.defaultAnnotationColors[tasks.indexOf(currentTask)]
+    }
+
+    return currentTask
+  }, [tasks, value])
+
+  const tagPosition = useMemo(
+    () => ({
+      minX: min(zone.map(({ x }) => x * imageWidth)),
+      minY: min(zone.map(({ y }) => y * imageHeight)),
+      maxX: max(zone.map(({ x }) => x * imageWidth)),
+    }),
+    [zone, imageWidth, imageHeight]
+  )
+
+  const styleZoom = useMemo(() => {
+    switch (true) {
+      case isHovered:
+        return { fill: task?.color, opacity: 0.3 }
+      case isPrediction:
+        return { fillPatternImage: patternIcon, opacity: 0.5, dash: [4, 4] }
+      case isPrefill:
+        return { fill: 'transparent', opacity: 0.5, dash: [4, 4] }
+
+      default:
+        return { fill: 'transparent', opacity: 1 }
+    }
+  }, [isHovered, task, isPrediction, isPrefill])
+
+  useEffect(() => {
+    if (isSelected && trRef.current && lineRef.current) {
+      trRef.current.nodes([lineRef.current])
+      trRef.current.getLayer().batchDraw()
+    }
+  }, [isSelected, trRef.current, lineRef.current])
+
+  const _onMouseEnter = () => setIsHovered(true)
+
+  const _onMouseLeave = () => setIsHovered(false)
+
+  if (!imageHeight && !imageWidth) {
+    return null
+  }
+
+  return (
+    <Group draggable={isEditable} onMouseEnter={_onMouseEnter} onMouseLeave={_onMouseLeave} {...props}>
+      <Group position="relative">
+        {!isEditable && (
+          <Image position="absolute" width={24} height={24} image={tagIcon} x={tagPosition.minX} y={tagPosition.minY} />
+        )}
+        {isHovered && (
+          <Text
+            position="absolute"
+            fontSize={16}
+            text={task?.label}
+            x={!isEditable ? tagPosition.minX + 30 : tagPosition.minX + 4}
```

Comment:
I tend to prefer to have the positive section in the first condition of binary because it's easier to read 
```
true ? trueCondition : falseCondition 
```